scripts/configChangePayloads/createSetQuorumSignatures.ts

```diff
@@ -0,0 +1,130 @@
+import { ethers } from 'ethers'
+import { GcpKmsSigner } from 'ethers-gcp-kms-signer'
+import fs from 'fs'
+import path from 'path'
+import { parse } from 'ts-command-line-args'
+
+import { getChainIdForNetwork } from '@layerzerolabs/lz-definitions'
+
+import { GcpKmsKey } from './kms'
+
+const PATH = path.join(__dirname)
+const FILE_PATH = `${PATH}/quorum-change-payloads.json`
+
+/**
+ * This script creates signature payloads to be submitted by an Admin of the DVN contract
+ * that will change the quorum of the DVN contract
+ */
+
+const args = parse({
+    environment: {
+        alias: 'e',
+        type: String,
+        defaultValue: 'mainnet',
+        description: 'environment',
+    },
+    chainNames: {
+        alias: 'c',
+        type: String,
+        description: 'comma separated list of chain names',
+    },
+    oldQuorum: {
+        type: Number,
+        description:
+            'old quorum, which is number of signatures required for change to happen',
+    },
+    newQuorum: {
+        type: Number,
+        description: 'new quorum',
+    },
+})
+
+const setQuorumFunctionSig = 'function setQuorum(uint64 _quorum)'
+const EXPIRATION = Date.now() + 7 * 24 * 60 * 60 * 1000 // 1 week expiration from now
+
+const iface = new ethers.utils.Interface([setQuorumFunctionSig])
+
+const getCallData = (newQuorum: number) => {
+    return iface.encodeFunctionData('setQuorum', [newQuorum])
+}
+
+const hashCallData = (target: string, callData: string, vId: string) => {
+    return ethers.utils.keccak256(
+        ethers.utils.solidityPack(
+            ['uint32', 'address', 'uint', 'bytes'],
+            [vId, target, EXPIRATION, callData],
+        ),
+    )
+}
+
+interface Signature {
+    signature: string
+    address: string
+}
+
+const main = async () => {
+    const { environment, chainNames, oldQuorum, newQuorum } = args
+    const dvnAddresses = require(`./data/dvn-addresses-${environment}.json`)
+    const keyIds = require(`./data/kms-keyids-${environment}.json`)
+    const signers = await Promise.all(
+        keyIds.map(async (credentials: GcpKmsKey) => {
+            return new GcpKmsSigner(credentials)
+        }),
+    )
+    const availableChainNames = chainNames.split(',')
+
+    const results: { [chainName: string]: any } = {}
+    await Promise.all(
+        availableChainNames.map(async (chainName) => {
+            results[chainName] = results[chainName] || {}
+            const vId = getChainIdForNetwork(chainName, environment, '2')
+            const callData = getCallData(newQuorum)
```

Comment:
a lot of duplicate code here that can probably benefit from just a function that iterates and generates signatures based on callbacks. i don't mind this for now but if we add more dvn signer functions this will get a bit error-prone to manual update.

Reply:
Refactored some of the duplicate code in the utils. Didn't do the callback approach because we want to put other info like vid and callData in the output file and having access them here is easier, but agree that's cleaner and can do that if you prefer that.